packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -290,6 +400,79 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
+        if (!opts || !opts.exclude) {
+            return;
+        }
+
+        const patternToPathMap: Map<string, string> = this.resolvePatternToPathMap(opts.exclude, rootPaths, true);
+        const updatedExcludes: string[] = [];
+        opts.exclude.forEach(pattern => {
+            const adjustedPattern = patternToPathMap.get(pattern) ? patternToPathMap.get(pattern) as string : pattern;
+            updatedExcludes.push(adjustedPattern);
+        });
+        opts.exclude = updatedExcludes;
+    }
+
+    /**
+     * The default search paths are set to be the root paths associated to a workspace
+     * however the search scope can be further refined with the include paths available in the search options.
+     * This method will replace the searching paths to the ones specified in the 'include' options but as long
+     * as the 'include' paths can be successfully validated as existing.
+     *
+     * Therefore the returned array of paths can be either the workspace root paths or a set of validated paths
+     * derived from the include options which can be used to perform the search.
+     *
+     * Any pattern that resulted in a valid search path will be removed from the 'include' list as it is
+     * provided as an equivalent search path instead.
+     *
+     * @param rootPaths - Workspace root paths
+     * @param opts
+     */
+    protected resolveSearchPathsFromIncludes(rootPaths: string[], opts: SearchInWorkspaceOptions | undefined): string[] {
+        if (!opts || !opts.include) {
+            return rootPaths;
+        }
+
+        const includesAsPaths = this.resolvePatternToPathMap(opts.include, rootPaths);
+        const patternPaths = Array.from(includesAsPaths.keys());
+
+        // Exclude include file patters that were successfully translated to search paths
+        opts.include = opts.include.filter(item => !patternPaths.includes(item));
+        return includesAsPaths.size > 0 ? Array.from(includesAsPaths.values()) : rootPaths;
+    }
+
+    /**
+     * Attempts to resolve valid file paths from a given list of patterns.
+     * The given (e.g. workspace) root paths are used to try resolving relative path patterns to an absolute path.
+     * The resulting map will include all patterns associated to its equivalent file path.
+     * The given patterns that are not successfully mapped to a file system path are not included.
+     *
+     * @param rootPaths
+     * @param patterns
+     */
+    protected resolvePatternToPathMap(patterns: string[], rootPaths: string[], keepSuffix: boolean = false): Map<string, string> {
+        const patternToPathMap = new Map<string, string>();
+
+        patterns.forEach(pattern => {
+            rootPaths.forEach(root => {
+                const foundPath = findPathInSuffixedPattern(root, pattern, keepSuffix);
+
+                if (foundPath) {
+                    patternToPathMap.set(pattern, foundPath);
+                }
+            });
+        });
```

Comment:
Does that logic mean that when multiple `foundPaths` are found for the same pattern, we'll only keep the last one?

Reply:
That's a good catch !
I have updated it so we can now resolve multiple paths for a single pattern, so the resulting map is now <string, string[]>
Done!